test/com/linkedin/parseq/TestTasks.java

```diff
@@ -89,6 +91,59 @@ public void onResolved(Promise<String> stringPromise) {
     assertEquals(Boolean.TRUE, resultRef.get());
   }
 
+  @Test
+  public void testFromFuture() {
+    String result = "FROMFUTURERESULT";
+    Task<String> task = Task.fromFuture(() ->{
+      CompletableFuture<String> completableFuture
+          = CompletableFuture.supplyAsync(() -> result);
+      return completableFuture;
+    });
+    String taskResult = runAndWait("testToTask.success", task);
+    assertEquals(result, taskResult);
+  }
+
+  @Test
+  public void testFromFutureWithTimeConsumingFuture() throws InterruptedException {
+    String result = "FROMFUTURERESULT";
+    Task<String> task = Task.fromFuture(() ->{
+      CompletableFuture<String> completableFuture
+          = CompletableFuture.supplyAsync(() -> {
+        try {
+          Thread.sleep(1000);
+        } catch (InterruptedException e) {
+          throw new RuntimeException(e);
+        }
+        return result;
+      });
+      return completableFuture;
+    });
+    run(task);
```

Comment:
Please use runAndWait instead.
I understand that you want to check if the task execution took at least 500ms but I think we can remove this part.